fairlearn/reductions/grid_search/grid_search.py

```diff
@@ -75,6 +108,57 @@ def fit(self, X, y, aux_data=None, **kwargs):
         # Prep the quality metric
         self.quality_metric.set_data(X, y_vector, A)
 
+        if isinstance(self.disparity_metric, Moment):
+            # Prep the disparity metric and objective
+            self.disparity_metric.init(X, A, y_vector)
```

Comment:
Should `self.disparity_metric` be turning into `self.constraint`  (although that can happen later)?